hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -208,4 +218,72 @@ public void disposeDeferredBlocks() {
     public Storage<Data, ? extends Record> getStorage() {
         return storage;
     }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccess;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdate;
+    }
+
+    @Override
+    public void increaseHits() {
+        this.hits++;
+    }
+
+    @Override
+    public void increaseHits(long hits) {
+        this.hits += hits;
+    }
+
+    @Override
+    public void decreaseHits(long hits) {
+        this.hits -= hits;
+    }
+
+    @Override
+    public void setLastAccessTime(long time) {
+        this.lastAccess = Math.max(this.lastAccess, time);
+    }
+
+    @Override
+    public void setLastUpdateTime(long time) {
+        this.lastUpdate = Math.max(this.lastUpdate, time);
+    }
+
+
+    protected void updateStatsOnPut(boolean newRecord, long now) {
+        setLastUpdateTime(now);
+
+        if (!newRecord) {
+            updateStatsOnGet(now);
```

Comment:
Makes sense but won't do it because we are trying to preserve existing behaviour.